base_geoengine/geo_db.py

```diff
@@ -18,22 +18,36 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
+"""Helper to setup Postgis"""
+from openerp.osv import osv
+from openerp.tools.translate import _
 
-from osv import fields, osv
+import logging
 
-from base_geoengine import geo_model
+logger = logging.getLogger('geoengine.sql')
 
 
-class SaleOrder(geo_model.GeoModel):
-    """Add geo_point to sale.order"""
-    _name = "sale.order"
-    _inherit = "sale.order"
-    _columns = {'geo_point' : fields.geo_related('partner_invoice_id',
-                                                 'geo_point',
-                                                 string='Sale order coordinate',
-                                                 type='geo_point',
-                                                 relation='res.partner.address',
-                                                 dim=2, # mandatory arg
-                                                 srid=900913)}# mandatory arg
-SaleOrder()
-
+def init_postgis(cursor):
+    """ Initialize postgis
+    """
+    cursor.execute("""
+        SELECT
+            tablename
+        FROM
+            pg_tables
+        WHERE
+            tablename='spatial_ref_sys';
+    """)
+    check = cursor.fetchone()
+    if check:
+        return {}
+    try:
+        cursor.execute("""
+        CREATE EXTENSION postgis;
+        CREATE EXTENSION postgis_topology;
+    """)
+    except Exception, exc:
+        raise osv.except_osv(
```

Comment:
These two lines are deprecated

It's recommended to use the `except Exception as exc` form with python2.6+
https://www.python.org/dev/peps/pep-3110/

osv is deprecated and orm.except_orm should be used (there are also v8 exceptions you can use, but I'm not too knowledgeable about them)